pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -981,24 +985,44 @@ func extractParentReferenceInfo(gateways map[parentKey][]*parentInfo, routeRefs
 			SectionName: ptr.OrEmpty(ref.SectionName),
 			Port:        ptr.OrEmpty(ref.Port),
 		}
+		gk := ir
+		if ir.Kind == gvk.Service || ir.Kind == gvk.ServiceEntry {
+			gk = meshParentKey
+		}
+		bannedHostnames := sets.New[string]()
```

Comment:
What's the point of this variable? seems like it gets shadowed by the other var on L997

Reply:
hmm, I had a lot of iterations. I think this is deadcode. Nice find